### PR TITLE
Disable pension and burial claims in production

### PR DESCRIPTION
**Pull request by rlbaker**
This PR will disable the `/v0/pension_claims` and `/v0/burial_claims` endpoints in production, while leaving them intact in test and development.

With this PR, `POST`-ing either of these endpoints in production will produce a 404 like the following:

```
HTTP/1.1 404 Not Found
Cache-Control: no-cache
Content-Type: application/json; charset=utf-8
Transfer-Encoding: chunked
Vary: Origin
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Git-SHA: f1af32135dd017276eab5f8931b0a9c92c0a95e8
X-GitHub-Repository: https://github.com/department-of-veterans-affairs/vets-api
X-Request-Id: e5c5e082-2455-44db-8bc3-0568151ef805
X-Runtime: 0.494378
X-XSS-Protection: 1; mode=block

{
    "errors": [
        {
            "code": "411",
            "detail": "There are no routes matching your request: v0/pension_claims",
            "status": "404",
            "title": "Not found"
        }
    ]
}
```

### CODE REVIEW

**Comment by rlbaker** [comment]
Merged in @wyattwalter's rubocop CVE fix so tests can run on this PR for now.

File: config/routes.rb [hunk 1]
```diff
@@ -42,8 +42,12 @@
         get(:healthcheck)
       end
     end
-    resource :pension_claims, only: [:create]
-    resource :burial_claims, only: [:create]
+
+    # Disabled in production until Central Mail API is in place
+    unless Rails.env.production?
```
Comment (lihanli) [inline config/routes.rb]: can you change this to use `Settings.pension_burial.upload.enabled`? staging and dev are both run with `RAILS_ENV=production` and we want to be able to test the central mail API there

Comment (rlbaker) [reply config/routes.rb]: Ah ok. Makes sense. Thought they'd be running under dev or test config. I'll fix that.

File: config/routes.rb [hunk 2]
```diff
@@ -42,8 +42,11 @@
         get(:healthcheck)
       end
     end
-    resource :pension_claims, only: [:create]
-    resource :burial_claims, only: [:create]
+
+    if Settings.pension_burial.upload.enabled?
```
Comment (lihanli) [inline config/routes.rb]: enabled shouldn't have a `?`

Comment (rlbaker) [reply config/routes.rb]: Yeah, just realized that. Pushed it up before my specs finished... Thanks for catching it.

**Pull request merged**